### PR TITLE
Sensu Go Enablement : Refactor for GA changes

### PR DESCRIPTION
**Pull request by jspaleta**
Refactor the Sensu Go enablement to account for GA changes.

## Description

* Check and Entity metadata was reorganized in beta8.

* beta8 also introduced new annotation concept to replace custom attributes. New mapping logic will treat env defined annotation as json string and will map into Sensu Core 1.x compatible Ruby hash representation of event data.

* Sensu Go versioning jumped to 5 from 2 in refactor. Enablement logic refactored  to indicate "Sensu Go" event mapping instead of specific event version. Function names were updated accordingly to reduce confusion.

* Added simple enablement mutator command that should help users start to migrate workloads. Provided mutator should help any workload that doesn't already require a mutator.

## Motivation and Context
Sensu Go beta8 refactor of Event model broke mapping to 1.x event.

## How Has This Been Tested?
Using updated handle-logevent.rb for manually updated sensu-plugins-logs plugin

Also updated the testing fixtures in the plugin to include changes

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

### CODE REVIEW

File: lib/sensu-handler.rb [hunk 1]
```diff
@@ -9,10 +9,10 @@ module Sensu
   class Handler
     include Sensu::Plugin::Utils
     include Mixlib::CLI
-    option :map_v2_event_into_v1,
-           description: 'Enable 2.x to 1.4 event mapping. Alternatively set envvar SENSU_MAP_V2_EVENT_INTO_V1=1.',
+    option :map_go_event_into_v1,
```
Comment (majormoses) [inline lib/sensu-handler.rb]: should we rename this to `map_go_event_into_ruby`? now that our hands are not tied we may very well have a 2.x ruby release.

File: lib/sensu-plugin/utils.rb [hunk 2]
```diff
@@ -74,15 +77,35 @@ def map_v2_event_into_v1(orig_event = nil)
...
           ##
           # Fill in renamed check attributes expected in 1.4 event
           #   subscribers, source
           ##
           event['check']['subscribers'] ||= event['check']['subscriptions']
-          event['check']['source'] ||= event['check']['proxy_entity_id']
+          event['check']['source'] ||= event['check']['proxy_entity_name']
 
           ##
           # Mimic 1.4 event action based on 2.0 event state
```
Comment (majormoses) [inline lib/sensu-plugin/utils.rb]: needs update

File: bin/mutator-sensu-go-into-v1.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env ruby
+
+require 'sensu-mutator'
+
+class Helper < Sensu::Mutator
+  def mutate
```
Comment (majormoses) [inline bin/mutator-sensu-go-into-v1.rb]: lets put a print statement in here to make it more obvious that this is only intended for development purposes.

Comment (jspaleta) [reply bin/mutator-sensu-go-into-v1.rb]: print to stderr?    I think the way mutators work is they output back to stdout.

File: lib/sensu-handler.rb (hunk 1 above)
Comment (jspaleta) [reply lib/sensu-handler.rb]: renamed in latest commit

File: lib/sensu-plugin/utils.rb (hunk 2 above)
Comment (jspaleta) [reply lib/sensu-plugin/utils.rb]: all references to 2.x and 1.x removed convered to Go and Ruby respectively in latest commit.

File: bin/mutator-sensu-go-into-v1.rb (hunk 3 above)
Comment (jspaleta) [reply bin/mutator-sensu-go-into-v1.rb]: STDERR.puts now in place with a warning message.  I also added a similar message in the mutator test to make sure it didn't screw anything up.

**Comment by jspaleta** [comment]
Latest commit should have addressed all outstanding comments.

File: bin/mutator-sensu-go-into-ruby.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env ruby
+
+require 'sensu-mutator'
+
+class Helper < Sensu::Mutator
+  def mutate
+    STDERR.puts 'WARNING: mutator-sensu-go-into-ruby.rb is meant for primarily for development.'
+    STDERR.puts 'Please update your handlers to be compatible with Sensu Go events.'
+    new_event = map_go_event_into_ruby
+    @event = new_event
```
Comment (portertech) [inline bin/mutator-sensu-go-into-ruby.rb]: Just single line it? `@event = map_go_event_into_ruby`

Comment (jspaleta) [reply bin/mutator-sensu-go-into-ruby.rb]: updated

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,18 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
+### Changes
+- renamed event mapping utility function to match naming change to Sensu Go map_go_event_into_ruby
```
Comment (majormoses) [inline CHANGELOG.md]: ideally we should indicate that
>  `map_v2_event_into_v1` was renamed to ...

Also these are breaking changes and should be called out as such.

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,18 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
+### Changes
```
Comment (majormoses) [inline CHANGELOG.md]: These are all breaking changes and I will fix this up post acceptance prior to release.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugin/versions/3.0.0

**Pull request merged**